### PR TITLE
feat: Sagemaker integration: `SagemakerGenerator`

### PR DESCRIPTION
**Pull request by ZanSara**
### Related Issues

- fixes https://github.com/deepset-ai/haystack-core-integrations/issues/275

### Proposed Changes:

- Add `SagemakerGenerator` with a very basic set of features (no streaming, no chat, only text inference)
- Add tests

### How did you test it?

Local unit and integration tests, unit tests in CI.

NOTE: integration tests won't run in the CI for now - keeping up the endpoint for these tests is very expensive.

### Notes for the reviewer

- [x] Tested on Falcon models
- [x] Tested on Llama2 models

As keeping Sagemaker endpoints up for testing is very expensive, integration tests are not going to run in CI.

### CODE REVIEW

File: integrations/amazon_sagemaker/src/haystack_integrations/components/generators/amazon_sagemaker/sagemaker.py [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+MODEL_NOT_READY_STATUS_CODE = 429
+
+
+@component
+class SagemakerGenerator:
+    """
+    Enables text generation using Sagemaker. It supports Large Language Models (LLMs) hosted and deployed on a SageMaker
+    Inference Endpoint. For guidance on how to deploy a model to SageMaker, refer to the
+    [SageMaker JumpStart foundation models documentation](https://docs.aws.amazon.com/sagemaker/latest/dg/jumpstart-foundation-models-use.html).
+
```
Comment (vblagoje) [inline integrations/amazon_sagemaker/src/haystack_integrations/components/generators/amazon_sagemaker/sagemaker.py:32]: Let's leave a note somewhere here that SageMaker doesn't support streaming (at least they didn't 6 months ago). Please check

File: integrations/amazon_sagemaker/src/haystack_integrations/components/generators/amazon_sagemaker/sagemaker.py [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+                list_output = [output]
+            elif isinstance(output, list) and all(isinstance(o, dict) for o in output):
+                list_output = output
+            else:
+                msg = f"Unexpected model response type: {type(output)}"
+                raise ValueError(msg)
+
+            # The key where the replies are stored changes from model to model, so we need to look for it.
+            # All other keys in the response are added to the metadata.
+            for key in self.model_generation_keys:
```
Comment (vblagoje) [inline integrations/amazon_sagemaker/src/haystack_integrations/components/generators/amazon_sagemaker/sagemaker.py:185]: Yeah, I think we can forget about matching the meta format here that we've strived to match so far. But hey, perhaps leave a code comment about that here...

File: integrations/amazon_sagemaker/src/haystack_integrations/components/generators/amazon_sagemaker/sagemaker.py [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+                region_name=os.getenv(self.aws_region_name_var),
+                profile_name=os.getenv(self.aws_profile_name_var),
+            )
+            self.client = session.client("sagemaker-runtime")
+        except Exception as e:
+            msg = (
+                f"Could not connect to SageMaker Inference Endpoint '{self.model}'."
+                f"Make sure the Endpoint exists and AWS environment is configured."
+            )
+            raise AWSConfigurationError(msg) from e
```
Comment (vblagoje) [inline integrations/amazon_sagemaker/src/haystack_integrations/components/generators/amazon_sagemaker/sagemaker.py:139]: Do we need to/from dict methods?

File: integrations/amazon_sagemaker/README.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+-----
+
+**Table of Contents**
+
+- [Installation](#installation)
+- [License](#license)
+
+## Installation
+
```
Comment (vblagoje) [inline integrations/amazon_sagemaker/README.md:14]: Why not match the other integration pages with hatch instructions?

**Review by vblagoje — CHANGES_REQUESTED** [review]
Minor changes and updates requested :-)

File: integrations/amazon_sagemaker/src/haystack_integrations/components/generators/amazon_sagemaker/sagemaker.py (hunk 1 above)
Comment (ZanSara) [reply integrations/amazon_sagemaker/src/haystack_integrations/components/generators/amazon_sagemaker/sagemaker.py:32]: Opened an issue for this feature, let's do it in the next iteration so at least we get basic support out of the door quickly https://github.com/deepset-ai/haystack-core-integrations/issues/285

**Review by vblagoje — APPROVED** [review]
Looks much better now, let's 🚢

**Pull request merged**